console_conf/controllers/identity.py

```diff
@@ -31,9 +31,37 @@ def default(self):
         self.ui.set_header(title, excerpt)
         self.ui.set_footer(footer, 40)
         self.ui.set_body(self.identity_view(self.model, self.signal, self.opts, self.loop))
+        if self.is_device_owned():
+            # FIXME: should be email rather than realname here, but extrausers only has email
+            self.signal.emit_signal('identity:done', self.model.user.realname)
+
+    def is_device_owned(self):
```

Comment:
The side-effecty nature of a method with this name is surprising. I get it now, but maybe it could be a get_device_owner() method that returns None if there is no user? And then we just run the next-screen signal, not done().